src/storage/test/UpdateVertexAndEdgeBenchmark.cpp

```diff
@@ -0,0 +1,416 @@
+}  // namespace nebula
+
+void updateVertex(int32_t iters, bool insertable) {
+    nebula::storage::cpp2::UpdateVertexRequest req;
+    BENCHMARK_SUSPEND {
+        req = nebula::storage::buildUpdateVertexReq(insertable);
+    }
+
+    for (decltype(iters) i = 0; i < iters; i++) {
+        // Test UpdateVertexRequest
+        auto* processor
+            = nebula::storage::UpdateVertexProcessor::instance(gKV.get(),
+                                                               schemaM.get(),
+                                                               indexM.get(),
+                                                               nullptr);
+        auto f = processor->getFuture();
+        processor->process(req);
+        auto resp = std::move(f).get();
+        if (!resp.result.failed_codes.empty()) {
+            LOG(ERROR) << "update faild";
+            return;
+        }
+    }
+}
+
+
+void updateEdge(int32_t iters, bool insertable) {
+    nebula::storage::cpp2::UpdateEdgeRequest req;
+    BENCHMARK_SUSPEND {
+        req = nebula::storage::buildUpdateEdgeReq(insertable);
+    }
+
+    for (decltype(iters) i = 0; i < iters; i++) {
+        // Test UpdateEdgeRequest
+        auto* processor
+             = nebula::storage::UpdateEdgeProcessor::instance(gKV.get(),
+                                                              schemaM.get(),
+                                                              indexM.get(),
+                                                              nullptr);
+        auto f = processor->getFuture();
+        processor->process(req);
+        auto resp = std::move(f).get();
+        if (!resp.result.failed_codes.empty()) {
+            LOG(ERROR) << "update faild";
+            return;
+        }
+    }
+}
+
+
+BENCHMARK(update_vertex, iters) {
+    updateVertex(iters, false);
+}
+
+BENCHMARK(upsert_vertex, iters) {
+    updateVertex(iters, true);
+}
+
+BENCHMARK(update_edge, iters) {
+    updateEdge(iters, false);
+}
+
+BENCHMARK(upsert_edge, iters) {
+    updateEdge(iters, true);
+}
+
+
+int main(int argc, char** argv) {
+    folly::init(&argc, &argv, true);
+    nebula::fs::TempDir rootPath("/tmp/UpdateTest.XXXXXX");
+    nebula::storage::setUp(rootPath.path());
+    folly::runBenchmarks();
+    gKV.reset();
+    return 0;
+}
+
+
+/**
+
+CPU : Intel(R) Xeon(R) CPU E5-2697 v3 @ 2.60GHz
+
+update_vertex   : tag data exist and update
+
+upsert_vertex   : tag data (first not exist) and upsert
+
+update_edge     : edge data exist and update
+
+upsert_edge     : edge data (first not exist)  and upsert
+
+V1.0
+==============================================================================
+src/storage/test/UpdateVertexAndEdgeBenchmark.cpprelative  time/iter  iters/s
+==============================================================================
+update_vertex                                               322.47us    3.10K
+upsert_vertex                                               324.23us    3.08K
```

Comment:
For upsert benchmark,  you'd better separate the cases "insert" and "update"

Reply:
Good suggestion, fixed.